src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -56,16 +66,18 @@ object DemuxFastqs {
   /** The name of the sample for unmatched reads. */
   val UnmatchedSampleId: String = "unmatched"
 
+  /** Contains the parameters used to control asynchronous computation and IO. */
+  private case class DemuxFastqsAsyncOptions(asyncReadingType: DemuxFastqsAsyncReadingType,
+                                             numWriterThreads: Int,
+                                             asyncDemux: Boolean,
+                                             demuxBatchSize: Int = 10e5.toInt) {
+    require(numWriterThreads >= 0)
+    require(!asyncDemux || demuxBatchSize > 0)
```

Comment:
Since you default to > 0, maybe remote the `!asyncDemux` as it's a little confusing.